lib/workers/repository/process/fetch.ts

```diff
@@ -17,27 +17,43 @@ import { Result } from '../../../util/result';
 import { LookupStats } from '../../../util/stats';
 import { PackageFiles } from '../package-files';
 import { lookupUpdates } from './lookup';
-import type { LookupUpdateConfig } from './lookup/types';
+import type { LookupUpdateConfig, UpdateResult } from './lookup/types';
 
-async function fetchDepUpdates(
-  packageFileConfig: RenovateConfig & PackageFile,
+type LookupResult = Result<PackageDependency, Error>;
+
+interface LookupTaskResult {
+  packageFile: PackageFile;
+  result: LookupResult;
+}
+
+type LookupTask = Promise<LookupTaskResult>;
+
+async function lookup(
```

Comment:
can you split these `Result` refactor to a separate PR? so this gets smaller and easier to review.

Reply:
This part is intact from the latest review, I just reverted the revert. The signature of function has changed, so I'll need modify upstream code just to match new signature, and then we'll remove it anyways. Let's not split this one.